content/en/blog/2023/kwok/index.md

```diff
@@ -0,0 +1,95 @@
+
+## What is KWOK?
+
+KWOK stands for Kubernetes WithOut Kubelet. So far, it provides two tools:
+
+`kwok`
+: `kwok` is the cornerstone of this project, responsible for simulating the lifecycle of fake nodes, pods, and other Kubernetes API resources.
+
+`kwokctl`
+: `kwokctl` is a CLI tool designed to streamline the creation and management of clusters, with nodes simulated by `kwok`.
+
+## Why use KWOK?
+
+KWOK has several advantages:
+
+- **Lightweight**: You can simulate thousands of nodes on your laptop without significant consumption of CPU or memory resources.
+- **Fast**: You can create and delete clusters and nodes almost instantly, without waiting for boot or provisioning.
+- **Compatibility**: [KWOK] works with any tools or clients that are compliant with Kubernetes APIs, such as kubectl, helm, etc.
+- **Portability**: [KWOK] has no specific hardware or software requirements. You can run it using prebuilt images, once Docker/Podman/Nerdctl is installed. Alternatively, binaries are also available for all platforms and can be easily installed.
+- **Flexibility**: You can configure different node types, labels, taints, capacities, conditions, etc., and you can configure different pod behaviors, status, etc. to test different scenarios and edge cases.
+
+{{< image width="90%"
+    link="./manage-clusters.svg"
+    caption="Using kwokctl to manage simulated clusters"
+    >}}
+
+## Testing Istio with KWOK
+
+For the latest information, see [KWOK with Istio].
+
+To create a test environment for the Istio control plane with [KWOK], you can follow these steps:
+
+### Set up Cluster
+
+{{< text bash >}}
+$ kwokctl create cluster --runtime kind
+{{< /text >}}
+
+### Create Node
+
+{{< text bash >}}
+$ kubectl apply -f https://kwok.sigs.k8s.io/examples/node.yaml
+{{< /text >}}
+
+### Deploy Istio
+
+{{< text bash >}}
+$ istioctl install -y
+{{< /text >}}
+
+### Migrate Controllers to Real Node
+
+We need to use [kind] to run the controller because [KWOK] itself cannot actually run a Pod.
+
+{{< text bash >}}
+$ kubectl patch deploy istiod -n istio-system --type=json -p='[{"op":"add","path":"/spec/template/spec/nodeName","value":"kwok-kwok-control-plane"}]'
+$ kubectl patch deploy istio-ingressgateway -n istio-system --type=json -p='[{"op":"add","path":"/spec/template/spec/nodeName","value":"kwok-kwok-control-plane"}]'
+{{< /text >}}
+
+### Create Pod and Inject Sidecar
+
+{{< text bash >}}
+$ kubectl label namespace default istio-injection=enabled
+$ kubectl apply -f {{< github_file >}}/samples/bookinfo/platform/kube/bookinfo.yaml
+{{< /text >}}
+
+You will see that the Pod is running on a fake node. However, it is still an injected sidecar.
+
+## Limitations
+
+Currently, we can only simulate the cluster data, but the real environment is actually connected with Sidecar's xDS, which cannot be simulated by [KWOK].
+
+So we need a [KWOK] based xDS client connect generator to complete the stress simulation, which is not yet available.
```

Comment:
I made a component [pwod](https://github.com/wzshiming/pwod) that generates xds connections based on pod data from a kwok cluster and it feels good!